### PR TITLE
Add geotarget filter, change log level to debug

### PR DESCRIPTION
**Pull request by scoursen**

### CODE REVIEW

File: octodns/provider/ns1.py [hunk 1]
```diff
@@ -204,16 +208,32 @@ def _params_for_A(self, record):
             # so that we know we did this on purpose if/when troubleshooting
             params['answers'] = [{"answer": [x], "meta": {}}
                                  for x in record.values]
+            has_country = False
             for iso_region, target in record.geo.items():
                 key = 'iso_region_code'
                 value = iso_region
+                if not has_country and len(value.split('-')) > 1:
+                    has_country = True
```
Comment (jroyalty) [inline octodns/provider/ns1.py:216]: Should this be reset to `False` within this loop?

File: octodns/provider/ns1.py [hunk 2]
```diff
@@ -204,16 +208,32 @@ def _params_for_A(self, record):
...
                 params['answers'].append(
```
Comment (jroyalty) [inline octodns/provider/ns1.py:217]: It seems like the presence of this inside this loop will re-add all answers that were inserted into `params['answers']` on line 209 in the previous version.

File: octodns/provider/ns1.py [hunk 3]
```diff
@@ -204,16 +208,32 @@ def _params_for_A(self, record):
...
                     {
                         'answer': target.values,
                         'meta': {key: [value]},
                     },
                 )
-        self.log.info("params for A: %s", params)
+            params['filters'] = []
+            if len(params['answers']) > 1:
```
Comment (jroyalty) [inline octodns/provider/ns1.py:224]: Should only add `shuffle` if more than one answer AND you have geo meta data on those answers.

**Review by jroyalty — CHANGES_REQUESTED** [review]
Just one change I think, and a couple of questions.

File: octodns/provider/ns1.py (hunk 1 above)
Comment (scoursen) [reply octodns/provider/ns1.py:216]: no, it starts as False, and flips to True if any of the answers for this record have the `iso_region_code` set.  as soon as that's present, we will need to set the `geotarget_country` filter.

File: octodns/provider/ns1.py (hunk 2 above)
Comment (scoursen) [reply octodns/provider/ns1.py:217]: the values in `record.values` are non-geo answers, so they are not redundant.

File: octodns/provider/ns1.py (hunk 3 above)
Comment (scoursen) [reply octodns/provider/ns1.py:224]: fixed.

**Pull request merged**